redux/cartSlice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const cartSlice = createSlice({
+  name: 'cart',
+  initialState: {
+    cart: <any>[],
+  },
+  reducers: {
+    addToCart: (state, action) => {
+      const itemInCart = state.cart.find((item: { id: any; }) => item.id === action.payload.id);
```

Comment:
Are your ids strings or numbers?  We should probably decide on a format (numbers are better) and type the `id` with that.